yt_dlp/extractor/drtuber.py

```diff
@@ -21,84 +23,201 @@ class DrTuberIE(InfoExtractor):
+            'duration': 421.0,
+            'age_limit': 18,
+            'like_count': int,
+            'dislike_count': int,
+            'comment_count': int,
+            'thumbnail': r're:https?://.+\.jpg',
+            'thumbnails': list,
+            'categories': list,
+        },
+    }, {
+        'url': 'http://m.nuvid.com/video/6415801/',
+        'md5': '638d5ececb138d5753593f751ae3f697',
+        'info_dict': {
+            'id': '6415801',
+            'display_id': '6415801',
+            'ext': 'mp4',
+            'title': 'My best friend wanted to fuck my wife for a long time',
+            'description': 'Welcome to this hot Redhead porn video named My Best Friend Wanted To Fuck My Wife For A Long Time. Nuvid is the best place for watching xxx movies online!',
+            'duration': 1882,
+            'age_limit': 18,
+            'like_count': int,
+            'dislike_count': int,
+            'comment_count': int,
+            'thumbnail': r're:https?://.+\.jpg',
+            'categories': list,
         },
     }, {
         'url': 'http://www.drtuber.com/embed/489939',
         'only_matching': True,
+    }, {
+        'url': 'https://www.iceporn.com/video/2296835',
+        'only_matching': True,
+    }, {
+        'url': 'https://www.nuvid.com/video/6513023',
+        'only_matching': True,
     }, {
         'url': 'http://m.drtuber.com/video/3893529/lingerie-blowjob-from-beautiful-teen',
         'only_matching': True,
     }]
 
     def _real_extract(self, url):
         mobj = self._match_valid_url(url)
-        video_id = mobj.group('id')
-        display_id = mobj.group('display_id') or video_id
+        video_id, display_id, domain = mobj.group('id', 'display_id', 'domain')
+        display_id = display_id or video_id
 
         webpage = self._download_webpage(
-            f'http://www.drtuber.com/video/{video_id}', display_id)
+            f'http://www.{domain}.com/video/{video_id}', display_id)
 
         video_data = self._download_json(
-            'http://www.drtuber.com/player_config_json/', video_id, query={
+            f'http://www.{domain}.com/player_config_json/', video_id, query={
                 'vid': video_id,
                 'embed': 0,
                 'aid': 0,
                 'domain_id': 0,
+            }, headers={
+                'Accept': 'application/json',
             })
 
+        qualities = {
+            'lq': '360p',
+            'hq': '720p',
+            '4k': '2160p',
+        }
+
         formats = []
         for format_id, video_url in video_data['files'].items():
             if video_url:
                 formats.append({
                     'format_id': format_id,
-                    'quality': 2 if format_id == 'hq' else 1,
+                    'quality': qualities.get(format_id) or format_id,
+                    'height': int_or_none(qualities.get(format_id)[:-1]),
```

Comment:
`quality` is for sorting purposes and needs to be `int` (default is `-1`)

use `utils.qualities`, e.g.:

```py
        QUALITIES = {
            'lq': 360,
            'hq': 720,
            '4k': 2160,
        }
        quality = qualities(tuple(QUALITIES))

        formats = []
        for format_id, video_url in video_data['files'].items():
            if video_url:
                formats.append({
                    'format_id': format_id,
                    'quality': quality(format_id),
                    'height': QUALITIES.get(format_id),
```